README.md
# trocalc
TalonRO calc


changelog.txt
[2020-07-20]
- Fixed Bragi formula
- Fixed MDEF armor enchantment
- Fixed Magmaring critical bonus
- Added Beach Towel, Cookie Hat and Black Tailed Ribbon
- Fixed Valorous Carnage Katar Sonic Blow damage bonus when linked

[2020-07-05]
- Fixed Health Point computation
- Fixed Triple Attack damage computation

[2020-07-04]
- Added Ancient Tower monsters
- Fixed The Ring of Nibelugen damage formula

[2020-06-21]
- Fixed AR/FAR ASPD party buff
- Fixed Eddga behaviour with MDEF
- Added new Elysian Garden monsters
- Shield Chain now take into consideration class defense bypass

[2020-05-29]
- Fixed Spiral Pierce damage computation

[2020-05-28]
- Form validation upon save action
- Fixed Sting Hat script
- Fixed Wood Goblin refine bonus damage
- Fixed ASPD value for Katar on Assassin
- Added Dragonfly Monocle

[2020-05-01]
- Fixed A Whistle Perfect Dodge bonus computation
- Fixed Stone Skin, Anti Magic, Magic Mirror and Keeping skills

[2020-04-26]
- Shield Chain now take into consideration race defense bypass
- Naming functionality for builds

[2020-04-24]
- Fixed Finger Offensive damage computation

[2020-04-19]
- Fixed Bragi behaviour
- Fixed hard DEF formula
- Added Candy Hat, Pitch Black Evil Druid Hat
- Fixed some monsters properties

[2020-04-11]
- Fixed aspd computation for double dagger
- Added Eclage food

[2020-04-04]
- Fixed Heal damage computation

[2020-04-02]
- Added Old Glast Heim cards 
- Added new Eclase's equipment
- Added Elysian Garden and Malangdo Culvert monsters

[2020-04-01]
- Added April patch cards 
- Added Eclase and Old Glast Heim monsters

[2020-03-25]
- Fixed Gold Scaraba card ATK bonus for damage computation
- Fixed Imperial Spear ATK bonus for damage computation
- Added Star Dust Hairband

[2020-03-24]
- Added Akkorokamui instance's equipment
- Removed MDEF penetration from Nightmare Verit card

[2020-02-29]
- Fixed flat MATK bonus for MATK computation
- Fixed magical elemental damage computation

[2020-01-05]
- Updated BG items

[2019-12-22]
- Brasilis and Moscovia cards

[2019-12-07]
- Fixed CRIT formula
- Fixed misc bugs

[2019-10-30]
- Added Old Glast Heim items

[2019-10-27]
- Fixed Damage calculation with Wakwak card
- Fixed Attack Modifier update with Bakonawa card
- Updated Elemental armors wearable by Novice/Super Novice
- Fixed Vanilla restrictions for Malaya MVP cards
- Added Flat MATK bonus for the card having this effect

[2019-10-12]
- Added - Ribbon Magician Hat


[2019-10-06]
- Added - Maylaya Cards [Hatfun] & [ZoneSoldier]
- Fixed - Imperial Spear add's 2 + atk per level of spear mastery now.[ZoneSoldier]
- Added - Increases magical damage against bosstype monsters [Hatfun]

[2018-08-11]
- Fixed Poison Knife itemid [Loa]
- Fixed FLEE reduction when mobbed [Loa]
- Fixed soft DEF reduction when mobbed [Loa]
		Combat calculations only, not display

[2018-08-09]
- Fixed Bow of Evil jobid [Loa]

[2018-08-08]
- Removed Piwik [Amor]

[2018-08-06]
- BG Item Proofreads [Loa]
		Medal of Honor (Thief)
		Brave Gladiator Blade
		Brave Battle Fist
		Brave Battle Strategy Book
		Assaulter Lance
		Speedy Recovery Wand
		Valorous Assassin Damascus
		Valorous Battle Fist
		Valorous Gladiator Blade
		Various set effects
		All Medal of Honor items
		Glorious Claymore
		Glorious Gladius

[2018-08-05]
- Added Angra Mantis Card [Loa]
- Fixed Drake Card skill [NattWara]

[2018-08-02]
- Fixed Sniper Cecil stats [NattWara]
- Fixed Monster FLEE up formula [NattWara]

[Update 2018-07-27]
- BG Items [Amor]
		Elite Engineer Armor
		Assassin Robe
		Warlock's Battle Robe
		Medic's Robe
		Elite Archer Suit
		Elite Shooter Suit
		Medalion [Merchant]
		Medalion [Acolyte]
		Medalion [Swordsman]
		Sheriff Badge
		Medalion [Mage/SL]
		Medalion [Thief]
		Medalion [Archer]
		Glorious Apocalipse
		Glorious Arc Wand
		Glorious Bloody Roar
		Glorious Claw
		Glorious Claymore
		Glorious Cleaver
		Glorious Fist
		Glorious Flamberge
		Glorious Gatiling Gun
		Glorious Gladius
		Glorious Grenade Launcher
		Glorious Guitar
		Glorious Holy Avenger
		Glorious Hunter Bow
		Glorious Huuma Shuriken
		Glorious Jamadhar
		Glorious Lance
		Glorious Lariat
		Glorious Morning Star
		Glorious Rapier
		Glorious Revolver
		Glorious Rifle
		Glorious Shotgun
		Glorious Spear
		Glorious Staff of Destruction
		Glorious Staff of Recovery
		Glorious Tablet
		Glorious Two Handed Axe
		Glorious Suit
		Glorious Shoes
		Glorious Muffler
		Glorious Ring

[Update 2018-07-26]
- BG Items [Amor]
	Soldier Grenade Launcher
	Soldier Rifle
	Soldier Shotgun
	Speedy Recovery Wand
	Valorous Assassin Damascus
	Valorous Assaulter's Katzbalger
	Valorous Battle CrossBow
	Valorous Battle Fist
	Valorous Battle Lariat
	Valorous Battle Strategy Book
	Valorous Battlefield Guitar
	Valorous Battlefield Morning Star
	Valorous Carnage Katar
	Valorous Gladiator Blade
	Valorous Huuma Front Shuriken
	Valorous Insane Battle Axe
	Warlock's Battle Wand
	Captain's Manteau
	Commander's Manteau
	Sheriff's Manteau
	Battle Greave
	Combat Boots
	Battle Boots
	Assaulter Plate

[Update 2018-07-25]
- Fixed an issue with def reduction formula when mobbed [Loa]
- Updated Items [Loa]:
	Brave Assaulter's Katzbalger
	Brave Battle CrowBow
	Brave Battle Fist
	Brave Battle Lariat
- Updated Battlegrounds Items [Amor]:
	Brave Battle Strategy Book
	Brave Battle Fist
	Brave Battlefield Guitar
	Brave Battlefield Morning Star
	Brave Carnage Katar
	Brave Gladiator Blade
	Brave Huuma Front Shuriken
	Brave Insane Battle Axe
	Soldier Gatling Gun
	Strong Recovery Wand
	Warlock's Magic Wand
	Assaulter Lance

[Update 2018-07-24]
- Fixed Eden weapon enchants 3rd slot adjustment according to 1st and 2st slot [NattWara]
- Fixed Army Padding not showing Mora enchants [NattWara]
- Fixed Golden Trickle 3rd enchant options not showing up correctly [Loa]
- Updated DEF reduction from being attacked by multiple monsters [Loa]
- Updated Items [Loa]:
	Assaulter Spear
	Brave Assassin Damascus

[Update 2018-07-23]
- Added n_tok for magical damage against a certain element (350-359) [Loa]

js/monster1.js
NameCalc = ["Base Stats",
"Auto Adjust Base Level",
"LV",
"JobLV",
"Class",
"STR",
"AGI",
"VIT",
"DEX",
"INT",
"LUK",
"Remaining Status Points",
"Extended Info",
"Max HP",
"Max SP",
"DEF",
"MDEF",
"HIT",
"FLEE",
"Perfect Dodge",
"Critical",
"MATK",
"ASPD",
"HP Regen",
"SP Regen",
"Speed Potion: ",
"Weapon Type: ",
"Weapon & Armor / Cards",
"Weapon: ",
"Attribute: ",
"Weapon: ",
"Demi-Human",
"Passive / Duration Skills",
0,
"Show",
"Attack Skills: ",
"Enemy: ",
"HP",
"LV",
"DEF",
"VIT",
"MDEF",
"INT",
"AGI",
"LUK",
"Race",
"Element",
"Size",
"Medium Size",
"Enemy Debuffs",
"Show",
"Combat Results",
"Hit Ratio",
0,
"Minimum Damage",
"Average Damage",
"Maximum Damage",
"Damage Per Second",
"Minimum Number of Hits",
"Average Number of Hits",
"Maximum Number of Hits",
"Average Battle Duration",
"WoE-Reduction",
0,
0,
0,
"%",
0,
0,
0,
0,
0,
0,
0,
0,
0,
"Minimum Delay Between Active Skills:",
"Adopted",
0,
"Item Data",
"Show",
"<Font size=2><B>Anything below this line will only be saved via URL</B></Font>",
"Enemy Buffs",
"Show",
];
for(i=1;i<=84;i++){
	if(NameCalc[i-1] != 0)
		myInnerHtml("nm0"+i,NameCalc[i-1],0);
}
MonsterOBJ = [
[0,7,7,1,1,90,11966,80,5,18,45,10,1,1,31,45,1,1,1,0,0,0,0,0,"Knight"],
[1,8,7,1,1,90,6990,40,90,6,30,30,1,1,26,30,1,1,1,0,0,0,0,0,"Assassin"],
[2,9,7,1,1,90,6955,95,5,105,25,10,1,1,30,40,1,1,1,0,0,0,0,0,"Priest"],
[3,10,7,1,1,90,4536,11,94,16,140,30,1,1,24,26,1,1,1,0,0,0,0,0,"Hunter"],
[4,11,7,1,1,90,3303,6,17,102,102,10,1,1,29,36,1,1,1,0,0,0,0,0,"Wizard"],
[5,12,7,1,1,90,7513,80,10,18,40,10,1,1,30,40,1,1,1,0,0,0,0,0,"Blacksmith"],
[6,13,7,1,1,90,11823,90,10,48,30,10,1,1,31,45,1,1,1,0,0,0,0,0,"Crusader"],
[7,14,7,1,1,90,7142,80,15,36,100,10,1,1,32,30,1,1,1,0,0,0,0,0,"Rogue"],
[8,15,7,1,1,90,7497,74,15,48,60,10,1,1,30,40,1,1,1,0,0,0,0,0,"Monk"],
[9,16,7,1,1,90,7174,100,15,66,80,10,1,1,33,30,1,1,1,0,0,0,0,0,"Barde"],
[10,17,7,1,1,90,7174,100,15,66,80,10,1,1,33,30,1,1,1,0,0,0,0,0,"Dancer"],
[11,18,7,1,1,90,6107,60,5,80,90,10,1,1,29,36,1,1,1,0,0,0,0,0,"Sage"],
[12,19,7,1,1,90,7513,80,5,18,40,10,1,1,30,40,1,1,1,0,0,0,0,0,"Alchemist"],
[13,20,7,1,1,1,40,1,1,1,1,1,1,1,0,0,1,1,1,0,0,0,0,0,"Super Novice"],
[14,21,7,1,1,90,15456,86,10,24,50,10,1,1,31,45,1,1,1,0,0,0,0,0,"Lord Knight"],
[15,22,7,1,1,90,9361,50,90,1,40,40,1,1,26,30,1,1,1,0,0,0,0,0,"Assassin Cross"],
[16,23,7,1,1,90,8917,100,20,111,20,10,1,1,30,40,1,1,1,0,0,0,0,0,"High Priest"],
[17,24,7,1,1,90,6150,24,106,12,140,30,1,1,24,26,1,1,1,0,0,0,0,0,"Sniper"],
[18,25,7,1,1,90,3936,15,10,105,100,10,1,1,29,36,1,1,1,0,0,0,0,0,"High Wizard"],
[19,26,7,1,1,90,9912,90,17,12,40,10,1,1,30,40,1,1,1,0,0,0,0,0,"Whitesmith"],
[20,27,7,1,1,90,14932,100,10,24,35,10,1,1,31,45,1,1,1,0,0,0,0,0,"Paladin"],
[21,28,7,1,1,90,9920,100,30,24,90,10,1,1,32,30,1,1,1,0,0,0,0,0,"Stalker"],
[22,29,7,1,1,90,10017,80,15,48,60,10,1,1,30,40,1,1,1,0,0,0,0,0,"Champion"],
[23,30,7,1,1,90,8467,100,20,70,80,10,1,1,33,30,1,1,1,0,0,0,0,0,"Clown"],
[24,31,7,1,1,90,8467,100,20,70,80,10,1,1,33,30,1,1,1,0,0,0,0,0,"Gypsy"],
[25,32,7,1,1,90,7133,60,15,80,90,10,1,1,29,36,1,1,1,0,0,0,0,0,"Scholar"],
[26,33,7,1,1,90,9912,90,10,18,40,10,1,1,30,40,1,1,1,0,0,0,0,0,"Creator"],
[27,0,7,1,1,1,40,1,1,1,1,1,1,1,0,0,1,1,1,0,0,0,0,0,"Novice"],
[28,1,7,1,1,1,40,1,1,1,1,1,1,1,0,0,1,1,1,0,0,0,0,0,"Swordsman"],
[29,2,7,1,1,1,40,1,1,1,1,1,1,1,0,0,1,1,1,0,0,0,0,0,"Thief"],
[30,3,7,1,1,1,40,1,1,1,1,1,1,1,0,0,1,1,1,0,0,0,0,0,"Acolyte"],
[31,4,7,1,1,1,40,1,1,1,1,1,1,1,0,0,1,1,1,0,0,0,0,0,"Archer"],
[32,5,7,1,1,1,40,1,1,1,1,1,1,1,0,0,1,1,1,0,0,0,0,0,"Mage"],
[33,6,7,1,1,1,40,1,1,1,1,1,1,1,0,0,1,1,1,0,0,0,0,0,"Merchant"],
[34,0,7,1,1,1,40,1,1,1,1,1,1,1,0,0,1,1,1,0,0,0,0,0,"High Novice"],
[35,1,7,1,1,1,40,1,1,1,1,1,1,1,0,0,1,1,1,0,0,0,0,0,"High Swordsman"],
[36,2,7,1,1,1,40,1,1,1,1,1,1,1,0,0,1,1,1,0,0,0,0,0,"High Thief"],
[37,3,7,1,1,1,40,1,1,1,1,1,1,1,0,0,1,1,1,0,0,0,0,0,"High Acolyte"],
[38,4,7,1,1,1,40,1,1,1,1,1,1,1,0,0,1,1,1,0,0,0,0,0,"High Archer"],
[39,5,7,1,1,1,40,1,1,1,1,1,1,1,0,0,1,1,1,0,0,0,0,0,"High Mage"],
[40,6,7,1,1,1,40,1,1,1,1,1,1,1,0,0,1,1,1,0,0,0,0,0,"High Merchant"],
];

EnemyNum = MonsterOBJ.length -1;

for(i=0; i<=EnemyNum; i++)
{
	document.calcForm.B_Enemy.options[i] = new Option(MonsterOBJ[i][24],i);
}